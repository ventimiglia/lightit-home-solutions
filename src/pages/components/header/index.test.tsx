import { render } from '@testing-library/react';
import Header from './';

describe('Header', () => {
  it('renders the header component', () => {
    const {container, getByAltText, getAllByRole} = render(<Header />);

    const logoElement = getByAltText('logo');
    expect(logoElement).toBeInTheDocument();

    const selectOptions = getAllByRole('option');
    expect(selectOptions).toHaveLength(3);
    expect(selectOptions[0]).toHaveTextContent('Guardar y Salir');
    expect(selectOptions[1]).toHaveTextContent('Salir Sin Guardar');
    expect(selectOptions[2]).toHaveTextContent('Guardar y Continuar');
    expect(container).toMatchSnapshot();
  });
});
