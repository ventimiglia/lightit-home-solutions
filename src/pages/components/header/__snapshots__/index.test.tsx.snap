// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders the header component 1`] = `
<div>
  <header
    class="bg-gray-dark"
  >
    <div
      class="flex justify-between items-center px-48 py-6 mx-auto"
    >
      <img
        alt="logo"
        src="logo.png"
      />
      <div
        class="relative"
      >
        <select
          class="py-3 pl-3 pr-14 text-primary rounded-md appearance-none text-sm focus:outline-none focus:border-none"
        >
          <option
            role="option"
          >
            Guardar y Salir
          </option>
          <option
            role="option"
          >
            Salir Sin Guardar
          </option>
          <option
            role="option"
          >
            Guardar y Continuar
          </option>
        </select>
        <div
          class="absolute top-0 right-0 flex items-center justify-center w-12 h-full pointer-events-none border-l-2 border-l-gray-medium"
        />
        <img
          class="absolute top-1/2 right-3 -rotate-90 -translate-y-1/2 pointer-events-none"
          src="chevron.png"
          width="25"
        />
      </div>
    </div>
  </header>
</div>
`;
